tests/backup/backup_share_test.go

```diff
@@ -1653,44 +1655,82 @@ var _ = Describe("{CancelClusterBackupShare}", func() {
 
 		Step("Validate that no groups or users have access to backups shared at cluster level", func() {
 			log.InfoD("Validate no groups or users have access to backups shared at cluster level")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
 			log.Infof("User chosen to validate no access - %s", chosenUser)
 			log.InfoD("Checking backups user [%s] has after revoking", chosenUser)
-			var userBackups []string
-			var err error
+			userBackups, err := GetAllBackupsForUser(chosenUser, commonPassword)
+			log.FailOnError(err, "Failed to get all backups for user - [%s]", chosenUser)
 			noAccessCheck := func() (interface{}, bool, error) {
-				// Enumerate all the backups available to the user
-				userBackups, err = GetAllBackupsForUser(chosenUser, commonPassword)
-				log.FailOnError(err, "Failed to get all backups for user - [%s]", chosenUser)
-				log.Infof("Backups user [%s] has access to - %v", chosenUser, userBackups)
 				if len(userBackups) > 0 {
-					return "", true, fmt.Errorf("waiting for all backup access - [%v] to be revoked for user = [%s]",
-						userBackups, chosenUser)
+					log.Infof("Backups user [%s] has access to - %v", chosenUser, userBackups)
+					for _, backupName := range userBackups {
+						backupUID, err := Inst().Backup.GetBackupUID(ctx, backupName, orgID)
+						log.FailOnError(err, fmt.Sprintf("Getting UID for backup %v", backupName))
+						backupInspectRequest := &api.BackupInspectRequest{
+							Name:  backupName,
+							Uid:   backupUID,
+							OrgId: orgID,
+						}
+						resp, err := Inst().Backup.InspectBackup(ctx, backupInspectRequest)
+						log.FailOnError(err, fmt.Sprintf("error inspecting backup %v", backupName))
+						deletePendingStatus := api.BackupInfo_StatusInfo_DeletePending
+						deletingStatus := api.BackupInfo_StatusInfo_Deleting
+						actual := resp.GetBackup().GetStatus().Status
+						reason := resp.GetBackup().GetStatus().Reason
+						if actual == deletePendingStatus || actual == deletingStatus {
+							log.Infof("Ignoring the backup from user access as the backup is in [%s] state ,Reason:[%s]", actual, reason)
+							RemoveElementByValue(&userBackups, backupName)
+							continue
+						} else {
+							return "", true, fmt.Errorf("waiting for backup access - [%v] to be revoked for user = [%s]",
```

Comment:
Add the status value also in this `Errorf` because if it is not in `DeletingPending` or `Deleting` state, we need to know which state it is in

Reply:
Added backup status.